src/UI/Implementation/Component/Button/Standard.php

```diff
@@ -7,4 +7,26 @@
 use ILIAS\UI\Component as C;
 
 class Standard extends Button implements C\Button\Standard {
+
+	/**
+	 * @var bool
+	 */
+	protected $anim = false;
+
+	/**
+	 * @inheritdoc
+	 */
+	public function withLoadingAnimation($anim) {
+		$this->checkBoolArg("state", $anim);
```

Comment:
"state" => "anim", the first argument is for better error messages for developers and just repeats the arguments name. It might have been possible to implement that with reflection or some other magic, but that seemed to be to expensive at the point I was implementing the stuff.

Reply:
[This can go away thanks to PHP 7](https://github.com/ILIAS-eLearning/ILIAS/blob/trunk/src/UI/ROADMAP.md#php-7-typehints-beginner-2h), btw, but this is a task for another day.

Reply:
fixed